From: Tomas Henzl <thenzl@redhat.com>
Date: Sun, 29 Aug 2010 15:49:12 -0400
Subject: [block] cciss: retry commands from within sendcmd_withirq
Message-id: <1283097002-3341-14-git-send-email-thenzl@redhat.com>
Patchwork-id: 27858
O-Subject: [RHEL6 PATCH 13/63] cciss: On unit attention retry commands from
	within sendcmd_withirq.
Bugzilla: 568830
RH-Acked-by: Neil Horman <nhorman@redhat.com>

On unit attention retry commands from within sendcmd_withirq.

diff --git a/drivers/block/cciss.c b/drivers/block/cciss.c
index 96d1eac..0592519 100644
--- a/drivers/block/cciss.c
+++ b/drivers/block/cciss.c
@@ -2342,6 +2342,8 @@ static int check_target_status(ctlr_info_t *h, CommandList_struct *c)
 				h->ctlr, c->Request.CDB[0],
 				c->err_info->SenseInfo[2]);
 		}
+		if (check_for_unit_attention(h, c))
+			return IO_NEEDS_RETRY;
 		break;
 	default:
 		printk(KERN_WARNING "cciss%d: cmd 0x%02x"
